app/src/main/kotlin/org/stypox/dicio/di/SttInputDeviceWrapper.kt

```diff
@@ -102,7 +97,7 @@ class SttInputDeviceWrapperImpl(
         return when (setting) {
             UNRECOGNIZED,
             INPUT_DEVICE_UNSET,
-            INPUT_DEVICE_VOSK -> VoskInputDevice(appContext, okHttpClient, localeManager)
+            INPUT_DEVICE_VOSK -> VoskInputDevice(appContext, okHttpClient, localeManager, settings)
```

Comment:
If the user changes the settings after this has been built, the old `settings` instance will remain. It's probably better to maintain the previous code for the rest of the class and instead pass `dataStore` here.

Reply:
Hmm, I was trying to avoid using `runBlocking` in `VoskInputDevice.startListening`. But that might just be my limited knowledge of Kotlin coroutines. I'll give it a try the way you recommend and see if it works.